Update cloud-native link to the new architecture guide (#16007)
dotnet/docs: docs/architecture/modernize-with-azure-containers/modernize-existing-apps-to-cloud-optimized/what-about-cloud-native-applications.md

The [Cloud Native Computing Foundation (CNCF)](https://www.cncf.io/) is a primary promoter of cloud-native principles. Microsoft is a [member of the CNCF](https://azure.microsoft.com/blog/announcing-cncf/).

> For detailed guidance on how to design and develop cloud-native applications, read the following free e-Books:

Reviewer: ```suggestion
For detailed guidance on how to design and develop cloud-native applications, read the following free e-books:
```

> *  [Architecting Cloud-Native .NET Applications for Azure](https://docs.microsoft.com/dotnet/architecture/cloud-native/introduction)

Reviewer: ```suggestion
* [Architecting Cloud-Native .NET Applications for Azure](../../cloud-native/introduction.md)
```

> *  [.NET Microservices: Architecture for containerized .NET applications](https://aka.ms/microservicesebook).

Reviewer: why not link to the book here as well to be consistent? Also, if the reader is reading this page in another language, you're sending them to an English PDF instead of the material in their language.

```suggestion
* [.NET Microservices: Architecture for containerized .NET applications](../../microservices/index.md).
```

Author: @nishanil asked at some point to use the aka link for download metrics.

Also, English is the only language that's being generated to PDF.

Will ping him directly regarding this one.

Reviewer 2: I agree with Maira's comments. Let's be consistent in the links.

The [Cloud Native Computing Foundation (CNCF)](https://www.cncf.io/) is a primary promoter of cloud-native principles. Microsoft is a [member of the CNCF](https://azure.microsoft.com/blog/announcing-cncf/).

> For a sample definition and for more information about the characteristics of cloud-native applications, see the section [Introduction to cloud-native applications](https://docs.microsoft.com/dotnet/architecture/cloud-native/introduction) from the [Architecting Cloud Native .NET Applications for Azure](https://docs.microsoft.com/dotnet/architecture/cloud-native/). For specific guidance from Microsoft about how to implement a cloud-native application, see [.NET microservices: Architecture for containerized .NET applications](https://aka.ms/microservicesebook).

Reviewer 2: ```suggestion
For detailed guidance on how to design and develop cloud-native applications, read the following free e-Books:

*  [Architecting Cloud-Native .NET Applications for Azure](https://docs.microsoft.com/dotnet/architecture/cloud-native/introduction)
*  [.NET Microservices: Architecture for containerized .NET applications](https://aka.ms/microservicesebook).
```
